x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java

```diff
@@ -237,31 +307,89 @@ public void testRemoteAccessForCrossClusterSearch() throws Exception {
-                        + " is unauthorized for user [local_search_user] with effective roles [local_search]"
-                        + " because no remote indices privileges apply for the target cluster"
-                )
+                sendRequestWith.equals("user")
+                    ? containsString(
+                        "action [indices:data/read/search] towards remote cluster [my_remote_cluster]"
+                            + " is unauthorized for user [local_search_user] with effective roles [local_search]"
+                            + " because no remote indices privileges apply for the target cluster"
+                    )
+                    : containsString(
+                        "action [indices:data/read/search] towards remote cluster [my_remote_cluster] "
+                            + "is unauthorized for API key id ["
+                            + localApiKeyId
+                            + "] of user [local_search_user] because no remote indices privileges apply for the target cluster"
+                    )
             );
 
             // Check that authentication fails if we use a non-existent API key
-            updateClusterSettings(Settings.builder().put("cluster.remote.my_remote_cluster.authorization", randomEncodedApiKey()).build());
+            updateClusterSettings(
+                Settings.builder()
+                    .put(
+                        "cluster.remote.my_remote_cluster.authorization",
+                        randomValueOtherThan(remoteAccessApiKeyId, () -> randomEncodedApiKey())
+                    )
+                    .build()
+            );
             final ResponseException exception4 = expectThrows(
                 ResponseException.class,
-                () -> performRequestWithRemoteAccessUser(new Request("GET", "/my_remote_cluster:index1/_search"))
+                sendRequestWith.equals("user")
+                    ? () -> performRequestWithRemoteAccessUser(new Request("GET", "/my_remote_cluster:index1/_search"))
+                    : () -> performRequestWithApiKey(new Request("GET", "/my_remote_cluster:index1/_search"), localApiKeyEncoded)
             );
-            assertThat(exception4.getResponse().getStatusLine().getStatusCode(), equalTo(401));
-            assertThat(exception4.getMessage(), containsString("unable to authenticate user"));
-            assertThat(exception4.getMessage(), containsString("unable to find apikey"));
+            if (sendRequestWith.equals("user")) {
+                assertThat(exception4.getResponse().getStatusLine().getStatusCode(), equalTo(401));
+                assertThat(exception4.getMessage(), containsString("unable to authenticate user"));
+                assertThat(exception4.getMessage(), containsString("unable to find apikey"));
+            } else {
+                assertThat(exception4.getResponse().getStatusLine().getStatusCode(), equalTo(403));
```

Comment:
Nit: I think we're conflating two things here: the case where the remote access API key is not found vs. the QC-side API key. Would test this for an existing QC-side API key with the right privileges. In that case we should get the same error (401) back from the FC since we used an API key that doesn't exist on the FC.